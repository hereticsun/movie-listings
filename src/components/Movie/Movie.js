import React, { Component } from 'react';
import { connect } from 'react-redux';

import styles from './Movie.module.css';
import { isArray } from 'util';

export class Movie extends Component {

  render() {
    const {movie, imagePath, genres} = this.props;
    const imageSrc = `${imagePath}${movie.poster_path}`;
    let movieGenres = [];
    let genreList = '';
    if(Boolean(genres) &&isArray(genres.genres)) {
      movie.genre_ids.map(id => {
        return movieGenres.push(genres.genres.find(genre => genre.id === id));
      });
      genreList = movieGenres.map(({ name }) => name).join(', ');
    }

    return (
      <li className={styles.movie}>
        <div className={styles.movieInfo}>
          <h3 className={styles.movieTitle}>{movie.title}</h3>
          <dl>
            <dt>Rating:</dt>
            <dd>{movie.vote_average}</dd>
            <dt>Genres:</dt>
            <dd data-id="genres">{genreList}</dd>
          </dl>
        </div>
        <img src={imageSrc} alt={movie.title} className={styles.moviePoster} />
      </li>
    );
  }
}

function mapStateToProps(state) {
  return {
    genres: state.genres,
  };
}

export default connect(mapStateToProps, {})(Movie);