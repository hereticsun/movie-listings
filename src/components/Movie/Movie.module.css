.movie {
  list-style: none;
  position: relative;
  width: 342px;
  height: 513px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.movieInfo {
  background: #3c3f42;
  background: rgba(60, 63, 66, 0.8);
  flex: 0 0 40%;
  z-index: 1;
}

.movieTitle {
  font-size: 24px;
  color: #fff;
}

.moviePoster {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
}