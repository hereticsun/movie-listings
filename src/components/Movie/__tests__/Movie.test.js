import React from 'react';
import { shallow } from 'enzyme';
import { Movie } from '../Movie';

describe(`Movie`, () => {
  const props = {
    movie: {
      id: '1',
      title: 'Toy Story 4',
    }
  };
  const element = shallow(<Movie {...props} />);

  it('displays the movie', () => {
    expect(element).toMatchSnapshot();
    expect(element.find('h3')).toHaveLength(1);
    expect(element.find('img')).toHaveLength(1);
  });
});
