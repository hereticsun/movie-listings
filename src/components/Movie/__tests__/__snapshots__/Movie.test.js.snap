// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie displays the movie 1`] = `ShallowWrapper {}`;
